src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -355,7 +385,15 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
       ReactFiberInstrumentation.debugTool.onWillBeginWork(workInProgress);
     }
     // See if beginning this work spawns more work.
-    let next = beginWork(current, workInProgress, nextPriorityLevel);
+    let next;
+    const isFailedWork = capturedErrors && capturedErrors.has(workInProgress);
```

Comment:
Can we fork `performUnitOfWork` into a separate function then call it synchronous in a while loop while there is still capturedErrors? That way we don't have to check this condition thousands of times per frame (and whatever else optimizations can be done as a result).